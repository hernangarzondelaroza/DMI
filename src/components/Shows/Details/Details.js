import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import ShowsService from "./../../../services/shows.service";

import { Container, Row, Col, Image, Button } from "react-bootstrap";

import "./Details.scss";

export default function Details() {
  let params = useParams();

  const [shows, setShows] = useState([]);
  const [show, setShow] = useState(undefined);

  useEffect(() => {
    ShowsService.getShows()
      .then((response) => setShows(response))
      .catch((error) => console.log("Error", error));
  }, []);

  useEffect(() => {
    if (shows.data) {
      shows.data.map((element) => {
        if (element.show.id == params.id) {
          setShow(element.show);
        }
      });
    }
  }, [shows, params]);

  useEffect(() => {
    console.log(show);
  }, [show]);

  return (
    <Container style={{ margin: "0 auto" }}>
      <Row className="justify-content-md-center">
        <Col>
          <h1>{show?.name}</h1>
          <Image
            className="detailsImage mx-auto d-block"
            src={show?.image?.medium}
          />
          <p
            dangerouslySetInnerHTML={{
              __html: show?.summary || "No description provided"
            }}
          />
          <Button variant="link" className="mx-auto d-block">
            <a href={show?.officialSite} alt={show?.officialSite}>
              {show?.officialSite}
            </a>
          </Button>
        </Col>
      </Row>
    </Container>
  );
}
