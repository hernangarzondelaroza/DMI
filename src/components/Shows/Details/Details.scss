h1, p {
  color: white;
  text-align: center;
}

p {
  margin: 1rem 0;
}

.detailsImage {
  margin: 0 auto;
  width: 18.75rem
}