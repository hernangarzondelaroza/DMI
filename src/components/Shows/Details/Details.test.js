import React from 'react';
import renderer from 'react-test-renderer';
import Details from './Details';

describe('Details', () => {
  it('Should take a snapshot', () => {
    const component = renderer
      .create(<Details />);
    expect(component).toMatchSnapshot();
  })
})