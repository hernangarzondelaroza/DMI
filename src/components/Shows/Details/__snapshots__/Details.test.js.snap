// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Should take a snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "margin": "0 auto",
    }
  }
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <h1 />
      <img
        className="detailsImage mx-auto d-block"
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "No description provided",
          }
        }
      />
      <button
        className="mx-auto d-block btn btn-link"
        disabled={false}
        type="button"
      >
        <a />
      </button>
    </div>
  </div>
</div>
`;
